content/guides/developer-guides/delfin/_index.md

```diff
@@ -38,17 +38,167 @@ Existing Delfin Drivers for reference:
 
+  ```python
+
+    def get_storage(self, context):
+        # Do something here
+        return {
+            'name': 'sample',
+            'description': 'sample',
+            'vendor': 'sample_vendor',
+            'model': 'sample_model',
+            'status': 'normal',
+            'serial_number': 123,
+            'firmware_version': '1.0.0',
+            'location': 'HK',
+            'total_capacity': 123,
+            'used_capacity': 0,
+            'free_capacity': 123,
+            'raw_capacity': 123,
+            'subscribed_capacity': 123
+        }
+
+    def list_storage_pools(self, ctx):
+        pool_list = []
+        for idx in range(3):
+            p = {
+                "name": "sample_pool_" + str(idx),
+                "storage_id": self.storage_id,
+                "native_storage_pool_id": "sample_original_id_" + str(idx),
+                "description": "Sample Pool",
+                "status": "normal",
+                "total_capacity": 123,
+                "used_capacity": 0,
+                "free_capacity": 123,
+            }
+            pool_list.append(p)
+        return pool_list
+
+    def list_volumes(self, ctx):
+      volume_list = []
+      for i in range(1, 5):
+          v = {
+              "name": "sample_vol_" + str(i),
+              "storage_id": self.storage_id,
+              "description": "Sample Volume",
+              "status": "normal",
+              "native_volume_id": "sample_original_id_" + str(i),
+              "wwn": "fake_wwn_" + str(i),
+              "total_capacity": 12,
+              "used_capacity": 2,
+              "free_capacity": 10,
+          }
+          volume_list.append(v)
+          return volume_list
+
+    def add_trap_config(self, context, trap_config):
+        pass
+
+    def remove_trap_config(self, context, trap_config):
+        pass
+
+    def parse_alert(self, context, alert):
+        pass
+
+    def clear_alert(self, context, alert):
+        pass
+
+    def list_alerts(self, context, query_para=None):
+        pass
+
+  ```
+
 * Ensure create storages API call from Delfin, can load the driver successfully.
 
+  ```bash
+  curl -H "Content-Type: application/json" -X POST http://localhost:8190/v1/storages -d '
+  {
+    "rest": {
+      "host": "127.0.0.1",
+      "port": "22",
+      "username": "username",
+      "password": "password",
+      "vendor": "sample_vendor",
+      "model": "sample_model"
+    }
+  }'
+
+  # Example Response:
+
+  '{"created_at": "2020-09-15T14:10:10.692091", "updated_at": null, "id": "849e0e90-35bd-4d23-8155-53a5a583bcbf", "name": "sample", "vendor": "sample_vendor", "description": "sample", "model": "sample_model", "status": "normal", "serial_number": 123, "firmware_version": "1.0.0", "location": "HK", "total_capacity": 123, "used_capacity": 0, "free_capacity": 123, "raw_capacity": 123, "subscribed_capacity": 123, "sync_status": "SYNCED", "deleted_at": null, "deleted": false}'
```

Comment:
Please format this to improve readability.

Reply:
This is the an example response from backend as it is, for reference purpose.